src/Components/Components/src/RenderTree/EventHandlerDelegateWithClosuresComparer.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Collections.Concurrent;
+using System.Reflection;
+using System.Linq.Expressions;
+
+namespace Microsoft.AspNetCore.Components.RenderTree
+{
+    internal static class EventHandlerDelegateWithClosuresComparer
+    {
+        private static readonly ConcurrentDictionary<Type, Func<object, object, bool>> ClosureComparerCache = new();
```

Comment:
This runs within the sync context, so I believe it can just be a regular dictionary.

Reply:
Are you sure for server side blazor? There can be multiple circuits. Also in .NET 5 Concurrent dictionary is lock-free for reading, so this is why the comparer is static, and not an instance on the diff context.

Reply:
@jspuij yeah, you a re right.

Reply:
@jspuij my comment was more targeted at avoiding the extra type so that it could be trimmed.

Reply:
How big will this get?

Reply:
I think a pretty large application has maybe 100s of components, with less than ten closures on average.